src/transformers/pipelines.py

```diff
@@ -2631,6 +2641,49 @@ def _concat_inputs_history(self, inputs: List[List[int]], histories: List[Option
 }
 
 
+def check_task(task: str) -> Tuple[Dict, Any]:
+    """
+    Checks an incoming task string, to validate it's correct and return the
+    default Pipeline and Model classes, and default models if they exist.
+
+    Args:
+        task (:obj:`str`):
+            The task defining which pipeline will be returned. Currently accepted tasks are:
+
+            - :obj:`"feature-extraction"`
+            - :obj:`"sentiment-analysis"`
+            - :obj:`"ner"`
```

Comment:
Should be deprecated (my opinion) as we're pushing for users to do `pipeline("text-classification", "bert-base-sentiment")` or `pipeline("token-classification", "user/spanish-ner")` or even better `TokenClassificationPipeline(model, tokenizer)`

Wdyt?

Reply:
(the website reasons in terms of pipeline type, not in terms of actual "nlp task")

cc also @mfuntowicz for discussion

Reply:
This comment actually comes from the `pipeline` docstring, so it should probably be updated too.

What you mean is that tasks `sentiment-analysis` and `ner` are deprecated in favor of `text-classification` and `token-classification` ?

Reply:
I would say so, and maybe we should even remove `SUPPORTED_TASKS` completely

Reply:
How come ? It's quite useful to determine the actual Pipeline that is going to be used....

Reply:
I mean that our current vision is more model-centric than task-centric – with a model implying a task.

I'd be curious to know the proportion of users of pipelines who spawn one with one of these identifiers, vs. a model identifier

Reply:
In general, I also think it would be much less painful and cleaner if we go for pipelines that are 1:1 linked to our model type classes (model-centric). As discussed a bit with @mfuntowicz, I think a good idea would be to base all pipelines on our "model types", *e.g.* `TokenClassification`, `CausalLM`, `ForConditionalGeneration` . To be more "user-friendly" we could just create some aliases `NamedEnitityRecoginitionPipeline = TokenClassificationPipeline`, but I think it would be better to have all the logic in `TokenClassification`.

Reply:
But that's probably out-of-scope for this PR and is more the vision of Pipelines v2 if I understood correctly (@mfuntowicz ).